### PR TITLE
win: free uv__loops once empty

### PR DESCRIPTION
**Pull request by cjihrig**
This commit moves the allocation of `uv__loops` into `uv__loops_add()` and frees all of the memory associated with `uv__loops` in `uv__loops_remove()` once the loop count reaches zero.

Fixes: https://github.com/libuv/libuv/issues/1252
Fixes: https://github.com/libuv/libuv/issues/1125

### CODE REVIEW

File: src/win/core.c [hunk 1]
```diff
@@ -77,17 +77,14 @@ static void uv__crt_invalid_parameter_handler(const wchar_t* expression,
 }
 #endif
 
-static uv_loop_t** uv__loops;
+static uv_loop_t** uv__loops = NULL;
```
Comment (bnoordhuis) [inline src/win/core.c]: Not necessary, it's already static.

File: src/win/core.c [hunk 2]
```diff
@@ -77,17 +77,14 @@ static void uv__crt_invalid_parameter_handler(const wchar_t* expression,
...
 #define UV__LOOPS_CHUNK_SIZE 8
 static uv_mutex_t uv__loops_lock;
 
 static void uv__loops_init() {
   uv_mutex_init(&uv__loops_lock);
-  uv__loops = uv__calloc(UV__LOOPS_CHUNK_SIZE, sizeof(uv_loop_t*));
-  if (!uv__loops)
-    uv_fatal_error(ERROR_OUTOFMEMORY, "uv__malloc");
   uv__loops_size = 0;
   uv__loops_capacity = UV__LOOPS_CHUNK_SIZE;
```
Comment (bnoordhuis) [inline src/win/core.c]: This should be zero.  You might as well remove it and the line above it.

File: src/win/core.c [hunk 3]
```diff
@@ -98,7 +95,12 @@ static int uv__loops_add(uv_loop_t* loop) {
 
   uv_mutex_lock(&uv__loops_lock);
 
-  if (uv__loops_size == uv__loops_capacity) {
+  if (uv__loops == NULL) {
```
Comment (bnoordhuis) [inline src/win/core.c]: You can drop this block if you ensure `uv__loops_capacity == 0` when `uv__loops == NULL`.

File: src/win/core.c (hunk 2 above)
Comment (cjihrig) [inline src/win/core.c]: Doesn't this represent the size that `uv__loops_size` can grow to before a realloc occurs?

Comment (cjihrig) [inline src/win/core.c]: Ah, nevermind. I think I see what you're getting at.

**Comment by cjihrig** [comment]
@bnoordhuis updated. Much simpler, thanks.

CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/232/

**Pull request merged**